src/main/scala/Compiler.scala

```diff
@@ -16,15 +16,17 @@ object Compiler {
     "Hoist memory reads" -> passes.HoistMemoryReads,
     "Sequentialize" -> passes.Sequentialize,
     "Lower unroll and bank" -> passes.LowerUnroll,
-    "Lower for loops" -> passes.LowerForLoops
+    "Lower for loops" -> passes.LowerForLoops,
+    "Hoist slow binops" -> passes.HoistSlowBinop
```

Comment:
Why not run this after hoist memory reads? Is it because `lower unroll` might generate expense expressions?

Reply:
yeah. I haven't actually verified this though. do you think it should be run before unroll?

Reply:
Unrolling "just" works with hoisting since it doesn't have any assumptions about the structure apart from "all memory reads are registered".

The only problem is that shrink views might generate `%` and `/` when the bank number is not a power of two. I think as long as you are not introducing too many `---`, it should be fine. I guess there needs to some `---` between the computation of the multiply and the use of the register.